File: Examples/test-suite/perl5/using_inherit_runme.pl
use strict;
use warnings;
use Test::More tests => 14;
BEGIN { use_ok('using_inherit') }
require_ok('using_inherit');

my $b = using_inherit::Bar->new();
is($b->test(3), 3, "Bar::test(int)");
is($b->test(3.5), 3.5, "Bar::test(double)");

$b = using_inherit::Bar2->new();
is($b->test(3), 6,"Bar2::test(int)");
is($b->test(3.5), 7.0, "Bar2::test(double)");

$b = using_inherit::Bar3->new();
is($b->test(3), 6, "Bar3::test(int)");
is($b->test(3.5), 7.0, "Bar3::test(double)");

$b = using_inherit::Bar4->new();
is($b->test(3), 6,"Bar4::test(int)");
is($b->test(3.5), 7.0, "Bar4::test(double)");

$b = using_inherit::Fred1->new();
is($b->test(3), 3, "Fred1::test(int)");
is($b->test(3.5), 7.0, "Fred1::test(double)");

$b = using_inherit::Fred2->new();
is($b->test(3), 3, "Fred2::test(int)");
is($b->test(3.5), 7.0, "Fred2::test(double)");

File: Examples/test-suite/lua/using_inherit_runme.lua
local v=require("using_inherit")

-- Lua uses number for both float and integer
-- Lua can not differentiate, C/C++ must differentiate function names

local b = v.Bar()
assert(b:test(3) == 3, "Bar::test(int)")
-- assert(b:test(3.5) == 3.5, "Bar::test(double)")

b = v.Bar2()
assert(b:test(3) == 6, "Bar2::test(int)")
-- assert(b:test(3.5) == 7, "Bar2::test(double)")

b = v.Bar3()
assert(b:test(3) == 6, "Bar3::test(int)")
-- assert(b:test(3.5) == 7, "Bar3::test(double)")

b = v.Bar4()
assert(b:test(3) == 6, "Bar4::test(int)")
-- assert(b:test(3.5) == 7, "Bar4::test(double)")

b = v.Fred1()
assert(b:test(3) == 3, "Fred1::test(int)")
-- assert(b:test(3.5) == 7, "Fred1::test(double)")

b = v.Fred2()
assert(b:test(3.5) == 7, "Fred2::test(double)")
